### ISSUE TITLE
Wrong methods suggested on class

### ISSUE THREAD

**Issue by hultberg** [Gert-dev/php-integrator-autocomplete-plus#59]
Hello!

The following screenshot is from atom where the autocomplete-plus package suggests wrong methods on MyObject. None of the methods in the list is in that class. More interesting is that all those methods are a part of the test suite.

### Versions
````
Atom    : 1.10.0-dev-67046ff
Electron: 0.37.8
Chrome  : 49.0.2623.75
Node    : 5.10.0
````

...I'm able to reproduce in atom beta.

![screen shot 2016-07-28 at 13 01 34](https://cloud.githubusercontent.com/assets/1257913/17210515/b95be726-54c3-11e6-8b69-3c078498150e.png)

**Comment by hultberg** [comment]
This might be related:
````
PHP sent back something unexpected. This is most likely an issue with your setup. If you're sure this is a bug, feel free to report it on the bug tracker.
 
→ /Users/edvin/.atom/packages/php-integrator-base/php/src/IndexDataAdapter.php:103 - Circular dependency detected from HbLib\Sql\HbSqlBase to HbLib\Sql\HbSqlBase!
````

What is interesting here is that the class `HbSqlBase` no longer exists in my project.

**Comment by Gert-dev** [comment]
I've also experienced the same today, but have yet to find a reliable way to reproduce it. It probably has something to do with the wrong cache being used in a corner case. If you have a reliable way to reproduce it, that would be a huge help. If not, I'll continue trying to manually reproduce it.

**Comment by Gert-dev** [comment]
@hultberg Might be a bit late, but do you remember if you ever did a force reindex for the project where you experienced this? I just experienced this again and traced it back to somehow the wrong methods having been cached for a class ID (but the database is, in fact, correct). This _should_ be impossible as it can never be the case in a database that another class used the unique primary key before the new one did. That is... _unless_ there happened to be an old cache item from before the force reindex that happened to match the new ID.

In any case, a fix I have to add is that the cache is invalidated whenever a force reindex occurs as all the ID's will reset in that case. I'm thinking (hoping) that this is also the problem here, the timestamps of the cache files seem to be consistent with this theory, at least.

**Comment by Gert-dev** [comment]
The fix is included in 1.2.0. I hope that permanently solves it.
